google/resumable_media/_helpers.py

```diff
@@ -164,3 +179,180 @@ def wait_and_retry(func, get_status_code, retry_strategy):
             return response
 
     return response
+
+
+def _get_crc32c_object():
+    """ Get crc32c object
+    Attempt to use the Google-CRC32c package. If it isn't available, try
+    to use CRCMod. CRCMod might be using a 'slow' varietal. If so, warn...
+    """
+    try:
+        import crc32c
+
+        crc_obj = crc32c.Checksum()
+    except ImportError:
+        try:
+            import crcmod
+
+            crc_obj = crcmod.predefined.Crc("crc-32c")
+            _is_fast_crcmod()
+
+        except ImportError:
+            raise ImportError("Failed to import either `google-crc32c` or `crcmod`")
+
+    return crc_obj
+
+
+def _is_fast_crcmod():
+    # Determine if this is using the slow form of crcmod.
+    nested_crcmod = __import__(
+        "crcmod.crcmod", globals(), locals(), ["_usingExtension"], 0,
+    )
+    fast_crc = getattr(nested_crcmod, "_usingExtension", False)
+    if not fast_crc:
+        warnings.warn(_SLOW_CRC32C_WARNING, RuntimeWarning, stacklevel=2)
+    return fast_crc
+
+
+def _get_metadata_key(checksum_type):
+    if checksum_type == "md5":
+        return "md5Hash"
+    else:
+        return checksum_type
+
+
+def prepare_checksum_digest(digest_bytestring):
```

Comment:
Several other helper functions that are similarly independent of internal specifics are public in this file so I decided this one should be intentionally public as well.

Reply:
If that's the case should the file name be renamed to `helpers.py` instead in the case an end-user wants to use them?

Reply:
I don't think that would be productive as it would commit us to a public interface and there is no expressed customer demand for that, but I wanted to follow the previous convention of not marking functions in this private module further private if they aren't extremely single-use and don't make strong assumptions about internal structure.